src/main/java/frc/robot/commands/Drive.java

```diff
@@ -1,46 +1,69 @@
 package frc.robot.commands;
 
 import frc.robot.subsystems.Encoders;
+import frc.robot.subsystems.Encoders.TurnEncoder;
 import frc.robot.subsystems.Motors;
 import frc.robot.subsystems.Motors.TurnMotor;
+import frc.robot.subsystems.Gyroscope;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj.Joystick;
 import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
 /** An example command that uses an example subsystem. */
 public class Drive extends Command {
   @SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
   private final Encoders encoders;
   private final Motors motors;
+  private final Gyroscope gyro;
   private final Joystick joystick;
   private PIDController bearingControllerFrontLeft;
   private PIDController bearingControllerFrontRight;
   private PIDController bearingControllerBackLeft;
   private PIDController bearingControllerBackRight;
 
-  //find a way to update the PID values
   public double kP;
   public double kI;
   public double kD;
+  public ShuffleboardTab tab;
+  public GenericEntry P;
+  public GenericEntry I;
+  public GenericEntry D;
+
   private double currentBearing;
   private double previousBearingGoal;
+  private double fieldOrientOffset;
 
 
   /**
    * Creates a new ExampleCommand.
    *
    * @param subsystem The subsystem used by this command.
    */
-  public Drive(Motors motors, Encoders encoders, Joystick stick) {
+  public Drive(Motors motors, Encoders encoders, Gyroscope gyro, Joystick stick) {
     this.encoders=encoders;
     this.motors=motors;
     joystick=stick;
+    this.gyro=gyro;
+
     // Use addRequirements() here to declare subsystem dependencies.
-    addRequirements(motors,encoders);
+    addRequirements(motors,encoders,gyro);
   }
 
   // Called when the command is initially scheduled.
   @Override
   public void initialize() {
+    //tuning the PID constants
```

Comment:
I'm not sure about this - it seems that you've got each coefficient in a separate Shuffleboard tab; saving them in DriveConstants ones as defaults would be useful too.